shared-bindings/_canio/CAN.c

```diff
@@ -258,15 +247,20 @@ STATIC mp_obj_t canio_can_restart(mp_obj_t self_in) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(canio_can_restart_obj, canio_can_restart);
 
-//|     def listen(self, filters: Optional[Sequence[Filter]]=None, *, timeout: float=10) -> Listener:
+//|     def listen(self, match: Optional[Sequence[Match]]=None, *, timeout: float=10) -> Listener:
 //|         """Start receiving messages that match any one of the filters.
+//|
 //|         Creating a listener is an expensive operation and can interfere with reception of messages by other listeners.
+//|
 //|         There is an implementation-defined maximum number of listeners and limit to the complexity of the filters.
-//|         If the hardware cannot support all the requested filters, a ValueError is raised.  Note that generally there are some number of hardware filters shared among all fifos.
-//|         A message can be received by at most one Listener.
+//|
+//|         If the hardware cannot support all the requested matches, a ValueError is raised.  Note that generally there are some number of hardware filters shared among all fifos.
+//|
+//|         A message can be received by at most one Listener.  If more than one listener matches a message, it is undefined which one actually receives it.
+//|
 //|         An empty filter list causes all messages to be accepted.
-//|         Timeout dictates how long readinto, read and next() will block.
-//|         Readinto will return false(), read will return None, and next() will raise StopIteration."""
+//|
+//|         Timeout dictates how long readinto, read and next() will block."""
```

Comment:
How do I never timeout? I want this without an outside loop:

```python
for message in can.listen():
    print(message)
```

Reply:
I can follow whatever value other APIs use (-1?  None? infinity?) to indicate a "never timeout" condition in the Listener constructor.

However, I think that this is probably not good style and wouldn't be usable in any but the simplest cases, because it never provides the opportunity to do anything else while waiting for a message.

Reply:
or I could return None (instead of raising StopIteration) in the case of timeout with 'for message in can.listen()'.  I was just trying to implement what I thought we had agreed (admittedly I initially drafted it and may have glossed over concerns) about how the potentially blocking operations would work.